### PR TITLE
OOP Advanced add Solution 1

### PR DESCRIPTION
**Pull request by oleksandr-saulko**

### CODE REVIEW

File: src/main/java/core/basesyntax/ColorSupplier.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class ColorSupplier {
+    public static final int RANDOM_RANGE = 5;
+    private Random indexOfColor = new Random();
```
Comment (MaliukDaria) [inline src/main/java/core/basesyntax/ColorSupplier.java:7]:
```suggestion
    private Random random = new Random();
```

File: src/main/java/core/basesyntax/ColorSupplier.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    private String[] colors;
+
+    {
+        colors = new String[]{"green", "yellow", "red", "blue", "black", "orange"};
+    }
+
+    public String getIndexOfColor() {
+        return colors[indexOfColor.nextInt(RANDOM_RANGE)];
```
Comment (MaliukDaria) [inline src/main/java/core/basesyntax/ColorSupplier.java:15]:
```suggestion
        return colors[indexOfColor.nextInt(colors.length)];
```

and so we could delete RANDOM_RANGE constant

File: src/main/java/core/basesyntax/FigureSupplier.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+public class FigureSupplier extends ColorSupplier {
+    public static final int FIGURE_COUNT = 5;
+    private Random random = new Random();
+    private ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        int figureNumber = random.nextInt(FIGURE_COUNT);
+        // generate a specific figure based on the `figureNumber` value
+        switch (figureNumber) {
```
Comment (MaliukDaria) [inline src/main/java/core/basesyntax/FigureSupplier.java:13]: lets make 5 private methods getRandomCircle(), getRandomSquare(), ... and call them in switch-case construction

File: src/main/java/core/basesyntax/FigureSupplier.java [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+                return new Rectangle(colorSupplier.getIndexOfColor(),
+                        random.nextInt(100), random.nextInt(100));
+            case 4:
+                return new RightTriangle(colorSupplier.getIndexOfColor(),
+                        random.nextInt(100), random.nextInt(100));
+            default:
+                return new IsoscelesTrapezoid(colorSupplier.getIndexOfColor(),
+                        random.nextInt(100), random.nextInt(100),
+                        random.nextInt(100));
+        }
```
Comment (MaliukDaria) [inline src/main/java/core/basesyntax/FigureSupplier.java:28]: lets move number 100 to constant

**Review by MaliukDaria — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/DrawFigure.java [hunk 5]
```diff
@@ -0,0 +1,5 @@
+package core.basesyntax;
+
+public interface DrawFigure {
```
Comment (MaliukDaria) [inline src/main/java/core/basesyntax/DrawFigure.java:3]:
```suggestion
public interface Drawlable {
```

as we can use this interface not only for Figures

File: src/main/java/core/basesyntax/FigureSupplier.java [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+
+    private RightTriangle getRandomRightTriangle() {
+        return new RightTriangle(colorSupplier.getIndexOfColor(),
+                random.nextInt(MAX_VALUE), random.nextInt(MAX_VALUE));
+    }
+
+    private IsoscelesTrapezoid getRandomIsoscelesTrapezoid() {
+        return new IsoscelesTrapezoid(colorSupplier.getIndexOfColor(),
+                random.nextInt(MAX_VALUE), random.nextInt(MAX_VALUE), random.nextInt(MAX_VALUE));
+    }
```
Comment (MaliukDaria) [inline src/main/java/core/basesyntax/FigureSupplier.java:34]: private methods should be after public as helper ones

File: src/main/java/core/basesyntax/FigureSupplier.java [hunk 7]
```diff
@@ -0,0 +1,52 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier extends ColorSupplier {
```
Comment (MaliukDaria) [inline src/main/java/core/basesyntax/FigureSupplier.java:5]: you should also implement getDefaultFigure() method (read the task)

**Review by MaliukDaria — CHANGES_REQUESTED** [review]
Task: 
In the main() method we need to create an array of figures (the size of array can be 3 or 6, it doesn't matter). The first half of figures in this array should be generated with random parameters. The other half of figures should have the same, default parameters.
I cant sww this functionality

**Comment by oleksandr-saulko** [comment]
> Task: In the main() method we need to create an array of figures (the size of array can be 3 or 6, it doesn't matter). The first half of figures in this array should be generated with random parameters. The other half of figures should have the same, default parameters. I cant sww this functionality

I apologize. I made this task last summer and got merged 2 forks. I didn't saw when the main method was replaced for the base version.

File: src/main/java/core/basesyntax/Main.java [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+    public static void main(String[] args) {
+        FigureSupplier figureSupplier = new FigureSupplier();
+        Figure[] figures = new Figure[]{
+                figureSupplier.getRandomFigure(),
+                figureSupplier.getRandomFigure(),
+                figureSupplier.getRandomFigure(),
+                figureSupplier.getDefaultFigure(),
+                figureSupplier.getDefaultFigure(),
+                figureSupplier.getDefaultFigure()
+        };
```
Comment (MaliukDaria) [inline src/main/java/core/basesyntax/Main.java:15]: lets make it in a loop?

**Review by MaliukDaria — CHANGES_REQUESTED** [review]

**Comment by MaliukDaria** [comment]
> > Task: In the main() method we need to create an array of figures (the size of array can be 3 or 6, it doesn't matter). The first half of figures in this array should be generated with random parameters. The other half of figures should have the same, default parameters. I cant sww this functionality
> 
> I apologize. I made this task last summer and got merged 2 forks. I didn't see when the main method was replaced for the base version.

Oh, i see) This part of task was added later 🙂

**Review by MaliukDaria — APPROVED** [review]
GJ

**Pull request closed without merging**